There is a compilation error in the file craft/src/vault/pgp_key.rs.

There is a compilation error in the file craft/src/vault/pgp_key.rs when using the smallvec library version 2.0.0-alpha.3. The file compiles normally with smallvec version 1.11.1, but in commit ce11941, it was changed to version 2.0.0-alpha.3. It is unclear whether this is an error or for some other reason.
error[E0308]: mismatched types
   --> src/vault/pgp_key.rs:32:41
    |
32  |         .preferred_symmetric_algorithms(smallvec![SymmetricKeyAlgorithm::AES256]);
    |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `SmallVec<[SymmetricKeyAlgorithm; 8]>`, found `SmallVec<SymmetricKeyAlgorithm, _>`
    |
    = note: `SmallVec<SymmetricKeyAlgorithm, _>` and `smallvec::SmallVec<[SymmetricKeyAlgorithm; 8]>` have similar names, but are actually distinct types

--- a/craft/Cargo.toml
+++ b/craft/Cargo.toml
@@ -13,10 +13,9 @@ regex = "1.10.3 "
 vcpkg = "0.2"
 
 [dependencies]
-pgp = "0.10.2"
-#smallvec = "2.0.0-alpha.3"
-smallvec = "1.11.1"
-rusty_vault = { git = "https://github.com/Tongsuo-Project/RustyVault.git", rev = "ad9f6043678b0fdb5325091b635ee8f2ee891a3d" }
+pgp = "0.11.0"
+smallvec = "1.13.1"
+rusty_vault = { git = "https://github.com/Tongsuo-Project/RustyVault.git", version = "0.1.0"}
 fs2 = "0.4.3"
 gettext-rs = "0.7.0"
 gettext-sys = "0.21.0"
@@ -30,6 +29,7 @@ serde_json = "1.0.110"
 go-defer = "0.1.0"
 hex = "0.4.3"
 tracing = "0.1.40"
+zeroize = "1.7.0"
 
 [target.'cfg(target_os = "windows")'.dependencies]
 winapi = { version = "0.3", features = ["minwindef", "winnt", "winnls","winuser","shellapi","securitybaseapi","handleapi", "winerror"] }

--- a/craft/src/vault/mod.rs
+++ b/craft/src/vault/mod.rs
@@ -10,6 +10,7 @@ use rusty_vault::{
     http::sys::InitResponse,
     storage::{barrier_aes_gcm, physical},
 };
+use zeroize::Zeroizing;
 
 pub mod command;
 pub mod crypt;
@@ -66,10 +67,7 @@ pub fn load_core(work_dir: &Path) -> (Arc<RwLock<Core>>, Config) {
 /// and writes the generated keys to a secrets file.
 pub fn init_rv_core(work_dir: Option<&Path>) {
     // Define paths for configuration and secrets
-    let path = match work_dir {
-        Some(path) => path,
-        None => Path::new(WORK_DIR_PATH_DEFAULT),
-    };
+    let path = work_dir.unwrap_or_else(|| Path::new(WORK_DIR_PATH_DEFAULT));
     // let path = ;
     let config_path = path.join("config.hcl");
     let secrets_path = path.join("secrets");
@@ -133,10 +131,7 @@ pub fn init_rv_core(work_dir: Option<&Path>) {
 /// after successful unsealing.
 pub fn unseal_rv_core(work_dir: Option<&Path>) -> (Arc<RwLock<Core>>, String) {
     // Define paths for secrets
-    let path = match work_dir {
-        Some(path) => path,
-        None => Path::new(WORK_DIR_PATH_DEFAULT),
-    };
+    let path = work_dir.unwrap_or_else(|| Path::new(WORK_DIR_PATH_DEFAULT));
     // let path = Path::new(WORK_DIR_PATH_DEFAULT);
     let secrets_path = path.join("secrets");
 
@@ -152,11 +147,13 @@ pub fn unseal_rv_core(work_dir: Option<&Path>) -> (Arc<RwLock<Core>>, String) {
     let init_response: InitResponse =
         serde_json::from_str(&fs::read_to_string(secrets_path).unwrap()).unwrap();
     let init_result = InitResult {
-        secret_shares: init_response
-            .keys
-            .iter()
-            .map(|key| hex::decode(key).unwrap())
-            .collect(),
+        secret_shares: Zeroizing::new(
+            init_response
+                .keys
+                .iter()
+                .map(|key| hex::decode(key).unwrap())
+                .collect(),
+        ),
         root_token: init_response.root_token,
     };
 

--- a/craft/src/vault/pgp_key.rs
+++ b/craft/src/vault/pgp_key.rs
@@ -26,7 +26,7 @@ pub fn generate_key_pair(primary_user_id: &str) -> Result<KeyPair, anyhow::Error
     let mut key_params = SecretKeyParamsBuilder::default();
     key_params
         .key_type(KeyType::Rsa(2048))
-        .can_create_certificates(false)
+        .can_certify(false)
         .can_sign(true)
         .primary_user_id(primary_user_id.into())
         .preferred_symmetric_algorithms(smallvec![SymmetricKeyAlgorithm::AES256]);
@@ -101,12 +101,12 @@ pub fn decrypt(armored: &str, seckey: &SignedSecretKey) -> Result<String, anyhow
 
 // Encrypt message from file, and write it to a MGS_FILE waiting for decrypt
 // Arguments: message, read from file; public key file path
-pub fn encrypt_message(msg: &str, pubkey: &str) -> Result<String> {
-    let (pubkey, _) = SignedPublicKey::from_string(pubkey)?;
+pub fn encrypt_message(msg: &str, pub_key: &str) -> Result<String> {
+    let (pub_key, _) = SignedPublicKey::from_string(pub_key)?;
     // Requires a file name as the first arg, in this case I pass "none", as it's not used typically, it's just meta data
     let msg = pgp::Message::new_literal("none", msg);
     // convert data from OpenPGP Message to string
-    let armored = generate_armored_string(msg, pubkey)?;
+    let armored = generate_armored_string(msg, pub_key)?;
 
     Ok(armored)
 }

